dropwizard-jdbi/src/main/java/io/dropwizard/jdbi/DBIFactory.java

```diff
@@ -53,7 +57,14 @@ public DBI build(Environment environment,
         final String validationQuery = configuration.getValidationQuery();
```

Comment:
The problem that Tomcat-DBCP, which we use, doesn't support handling network
failures. Checkout this [article](https://github.com/brettwooldridge/HikariCP/wiki/Bad-Behavior:-Handling-Database-Down).